src/gear/simplifier.cc

```diff
@@ -165,74 +190,63 @@ an<Translation> Simplifier::Apply(an<Translation> translation,
   return New<SimplifiedTranslation>(translation, this);
 }
 
+void Simplifier::PushBack(const an<Candidate>& original,
+                         CandidateQueue* result, const string& simplified) {
+  string tips;
+  string text;
+  if (show_in_comment_) {
+    text = original->text();
+    tips = simplified;
+    comment_formatter_.Apply(&tips);
+  } else {
+    size_t length = utf8::unchecked::distance(original->text().c_str(),
+                                              original->text().c_str()
+                                              + original->text().length());
+    text = simplified;
+    if ((tips_level_ == kTipsChar && length == 1) || tips_level_ == kTipsAll) {
+      tips = original->text();
+      bool modified = comment_formatter_.Apply(&tips);
+      if (!modified) {
+        tips = quote_left + original->text() + quote_right;
+      }
+    }
+  }
+  result->push_back(
+      New<ShadowCandidate>(
+          original,
+          "simplified",
+          text,
+          tips));
+}
+
 bool Simplifier::Convert(const an<Candidate>& original,
                          CandidateQueue* result) {
   if (excluded_types_.find(original->type()) != excluded_types_.end()) {
     return false;
   }
-  size_t length = utf8::unchecked::distance(original->text().c_str(),
-                                            original->text().c_str()
-                                            + original->text().length());
-  bool success;
+  bool success = false;
   vector<string> forms;
-  success = opencc_->ConvertWord(original->text(), &forms);
-  if (success && forms.size() > 0) {
+  if (!random) success = opencc_->ConvertWord(original->text(), &forms);
```

Comment:
The control flow is hard to understand.
It doesn't work for human reader. Please have a separate branch / function for random conversion.